### DISCUSSION TITLE
CubeSatSim Receiving Data from Transmitter

### DISCUSSION THREAD

**Discussion by blueed9696** [alanbjohnston/CubeSatSim#342] [Q&A]
Hello Alan, 

I haven’t started on the receiving end yet, as I’ve been focused on transmitting bits using BPSK modulation. I’ll begin working on receiving data from CubeSatSim soon. Before I start, I’m wondering—if CubeSatSim has only one available serial port, is it possible to receive data from an external transmitter?

Thanks.

- Jay

**Comment by alanbjohnston** [comment]
Hi Jay,

The Raspberry Pi Pi Zero has only one serial port, and that serial port is connected to the Raspberry Pi Pico on the STEM Payload board.  The Raspberry Pi Pico has a second serial port implemented (J12 on the STEM Payload PCB) with software which can be connected to a serial GPS module  That serial port could connect to another transmitter with a serial output.

If the transmitter had an I2C or SPI interface, either the Pico or the Pi Zero could also connect to it.

Others have connected a LoRa transceiver to the CubeSatSim, but they haven't shared the details.

Alan
